jsonschema2pojo-integration-tests/pom.xml

```diff
@@ -102,6 +102,10 @@
             <version>${project.version}</version>
             <type>maven-plugin</type>
         </dependency>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-plugin-api</artifactId>
+        </dependency>
```

Comment:
Sorry if this I have missed something obvious, but is this needed?

Reply:
That's a great question @joelittlejohn and a place where a decision can be made.
With proposed solution this is needed as there were changes in `pom.xml`:
```xml
                <groupId>org.apache.maven</groupId>
                <artifactId>maven-plugin-api</artifactId>
                <version>2.2.1</version>
                <scope>provided</scope>
```
specifically `<scope>provided</scope>` was added to avoid:

> The following dependencies are in wrong scope:
> * org.apache.maven:maven-plugin-api:jar:2.2.1:compile

Which in turn made `maven-plugin-api` no longer available/visible to `jsonschema2pojo-integration-tests`.
There are several classes (`AnnotationStyleIT`, `CustomAnnotatorIT`, `CodeGenerationHelper`) in `jsonschema2pojo-integration-tests` that depend on `org.apache.maven.plugin.MojoExecutionException` (`maven-plugin-api`)
<br>
It is possible to remove _provided_ scope from `maven-plugin-api` though that would make warning reappear.
What do you think, should the _provided_ scope be removed or not ?

Reply:
Ah I see. Yes, that makes sense. I think it's good that because this dependency is now `provided` in the jsonschema2pojo-maven-plugin, the integration tests must declare this dependency explicitly.

This explicit dependency should have arguably always been there in the pom.xml for the integration tests. I don't like to rely on a transitive dependency if the dependency is in fact direct.